tests/end_to_end_test/end_to_end_test.go

```diff
@@ -553,6 +554,149 @@ func TestSnapshotDeleteTypeCheck(t *testing.T) {
 	e.runAndExpectFailure(t, "snapshot", "delete", manifestID, "--unsafe-ignore-source")
 }
 
+func TestSnapshotFail(t *testing.T) {
+	e := newTestEnv(t)
+	defer e.cleanup(t)
+	defer e.runAndExpectSuccess(t, "repo", "disconnect")
+
+	e.runAndExpectSuccess(t, "repo", "create", "filesystem", "--path", e.repoDir)
+	e.runAndExpectSuccess(t, "policy", "set", "--global", "--keep-latest", strconv.Itoa(1<<31-1))
+	source := filepath.Join(e.dataDir, "source")
+
+	// Test snapshot of nonexistent directory fails
+	e.runAndExpectFailure(t, "snapshot", "create", source)
+
+	testDirDepth := 2
+	createDirectory(t, source, testDirDepth)
+
+	// Create snapshot
+	e.runAndExpectSuccess(t, "snapshot", "create", source)
+
+	numSuccessfulSnapshots := 1
+
+	// Test the root dir permissions
+	fi, err := os.Stat(source)
+	if err != nil {
```

Comment:
You can leverage `assert` in tests to reduce the error check blocks

```suggestion
	assert.NoError(t, err, "could not state source:", source)
```

Reply:
Great call, forgot about that

Reply:
If you need to use `assert.X` multiple times, then the following helps as well:

```go
check := assert.New(t)

// ...
check.NoError(err, "message")

// ...
check.Equals(want, got, "message")

// ... etc
``

Reply:
oh most of the e2e test cases are using `assertNoError(t *testing.T, err error)` instead of the assert package. I ended up replacing the err checks with that.